src/js/auth.js

```diff
@@ -55,67 +92,56 @@ function authenticate(success) {
  * And authenticating authorization if auth0 redirected to AUTHD_PATH
  */
 function login(success) {
-  if (!navigator.onLine) {
-    success({ name: config.name });
+  if (appConfig.cypress) {
+    setToken(appConfig.cypress);
+    success();
     return;
   }
 
-  const AUTHD_PATH = '/authenticated';
-  config.redirect_uri = location.origin + AUTHD_PATH;
-  config.audience = 'care-ops-backend';
-
-  if (localStorage.getItem(RWELL_KEY)) {
-    config.connection = rwConnection;
+  if (!navigator.onLine) {
+    success();
+    return;
   }
 
-  const clientConfig = {
-    authorizationParams: config,
-    ...pick(config, 'clientId', 'domain'),
-    cacheLocation: _DEVELOP_ ? 'localstorage' : 'memory',
-  };
-
-  createAuth0Client(clientConfig).then(auth0Client => {
-    auth0 = auth0Client;
-    return auth0.isAuthenticated();
-  }).then(authed => {
-    if (location.pathname === '/logout') {
-      logout();
-      return;
-    }
-
-    // RWell specific login
-    if (location.pathname === `/${ RWELL_KEY }`) {
-      loginWithRedirect({
-        appState: RWELL_KEY,
-        authorizationParams: {
-          connection: rwConnection,
-        },
-      });
-      return;
-    }
-
-    if (location.pathname === AUTHD_PATH) {
-      authenticate(success).catch(() => {
-        forceLogin();
-      });
-      return;
-    }
-
-    if (!authed) {
-      forceLogin(location.pathname);
-      return;
-    }
-
-    if (location.pathname === '/login') {
-      window.history.replaceState({}, document.title, '/');
-    }
-
-    success({ name: config.name });
-  });
+  createAuth0Client(getConfig())
+    .then(setAuth0)
+    .then(isAuthenticated => {
+      if (location.pathname === '/logout') {
+        const federated = Radio.request('bootstrap', 'setting', 'federated_logout');
```

Comment:
Moved the `federated` setting to the users action of logout instead of doing it automatically in any logout..